python/lsst/obs/base/ingest_tests.py

```diff
@@ -150,34 +156,47 @@ def verifyIngest(self, files=None, cli=False, fullCheck=False):
         This only really affects files that contain multiple datasets.
         """
         butler = Butler(self.root, run=self.outputRun)
-        datasets = list(butler.registry.queryDatasets("raw", collections=self.outputRun))
+        datasets = list(butler.registry.queryDatasets(self.ingestDatasetTypeName, collections=self.outputRun))
         self.assertEqual(len(datasets), len(self.dataIds))
 
         # Get the URI to the first dataset and check it is inside the
         # datastore
         datasetUri = butler.getURI(datasets[0])
         self.assertIsNotNone(datasetUri.relative_to(butler.datastore.root))
 
+        # Get the relevant dataset type
+        datasetType = butler.registry.getDatasetType(self.ingestDatasetTypeName)
+
         for dataId in self.dataIds:
+
+            # Raws can be large so we don't want to read them every time
+            # but for now read the full dataset if it's not an Exposure
+
```

Comment:
I think this is a comment attached to the `if` below, so remove the space? But I'm not sure quite what to make of it: the comment suggests "we don't want to read raws, but we read them anyway?" Possibly a phrasing change?

Reply:
I'm saying that the else path only reads one Exposure and skips all the ohter reads but for non-Exposure we read all of it because we can't really know what components we are going to get. Will try to rephrase.